dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,147 @@
+import logging
+from datetime import datetime
+
+from analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+from analyzer_helper.discord.utils.is_user_bot import UserBotChecker
+
+
+class DiscordTransformRawData(TransformRawDataBase):
+
+    def __init__(self):
+        """
+        Initializes the class with a bot checker.
+        """
+        self.user_bot_checker = UserBotChecker()
+
+    def create_interaction(self, interaction_type, users_engaged_id, interaction_role):
+        return {
+            "name": interaction_type,
+            "users_engaged_id": users_engaged_id,
+            "type": interaction_role,
+        }
+
+    def create_receiver_interaction(self, data, interaction_type, engaged_user, author):
+        is_bot = self.user_bot_checker(author)
+        return {
+            "author_id": engaged_user,
+            "date": data["createdDate"]["$date"],
+            "source_id": data["messageId"],
+            "metadata": {
+                "thread_id": data["threadId"],
+                "channel_id": data["channelId"],
+                "bot_activity": data["isGeneratedByWebhook"] or is_bot,
+            },
+            "actions": [],
+            "interactions": [
+                {
+                    "name": interaction_type,
+                    "users_engaged_id": [author],
+                    "type": "receiver",
+                }
+            ],
+        }
+
+    def create_emitter_interaction(
+        self, user, period, data, interaction_type, engaged_user
```

Comment:
Please consider doing the following updates

1. Parameters type annotation missing.
2. the `engaged_user` could be a list of user ids, so please update it to a more appropriate name and consider updating the is_bot checking on line 47
3. I see a lot of similarities between the two functions `create_emitter_interaction` and `create_receiver_interaction`. Is it possible for you to merge the two into one function? (input parameters can change)